quantum/split_common/transaction_id_define.h

```diff
@@ -109,6 +109,10 @@ enum serial_transaction_id {
     SPLIT_TRANSACTION_IDS_USER,
 #endif // SPLIT_TRANSACTION_IDS_USER
 
+#if defined(SPLIT_DETECTED_OS_ENABLE)
```

Comment:
Also, this should go before the kb+users transactions, I think. 
```suggestion
#if defined(OS_DETECTION_ENABLE) && defined(SPLIT_DETECTED_OS_ENABLE)
```